pkg/gds/gds.go

```diff
@@ -112,6 +114,9 @@ func (s *GDS) Shutdown() (err error) {
 // Register generates a PKCS12 password, provided in the RPC response which can be
 // used to access the certificate private keys when they're emailed.
 func (s *GDS) Register(ctx context.Context, in *api.RegisterRequest) (out *api.RegisterReply, err error) {
+	// send register request activity to network activity handler
+	go activity.Register().Add()
```

Comment:
It shouldn't be necessary to run this as a go routine anymore, what the `Add()` does is just writes the activity to a channel. That should now be a non-blocking thing now because I've modified the method to just drop the event if the channel buffer is full.

Reply:
got it - I have made the change